spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/BasicAuthSecurityConfiguration.java

```diff
@@ -157,6 +161,9 @@ protected void configure(HttpSecurity http) throws Exception {
 		http.addFilterBefore(sessionRepositoryFilter,
 				ChannelProcessingFilter.class).csrf().disable();
 		http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.IF_REQUIRED);
+
+		securityStateBean.setAuthenticationEnabled(true);
+		securityStateBean.setAuthorizationEnabled(true);
```

Comment:
@jvalkeal Regarding `securityStateBean` - I was not to keen on using that BUT somehow I have to share some state that answers the basic questions:

* Is authentication enabled?
* Is authorization enabled?

As our security enablement is `implicit` and `conditional`, it gets a bit blurry to answer that question with Spring Boot properties alone. Also, our OAuth2 setup does not support roles directly as of yet. Therefore, I felt more comfortable to answer those 2 question from within  the configuration classes `BasicAuthSecurityConfiguration` and `OAuthSecurityConfiguration` directly.